perf-event/examples/cpufreq.rs

```diff
@@ -0,0 +1,54 @@
+use std::process::ExitCode;
+use std::time::Duration;
+
+use perf_event::events::MSRConfig;
+use perf_event::events::MSREvent;
+use perf_event::Builder;
+
+fn run() -> std::io::Result<()> {
+    let tsc_event = MSREvent::with_config(MSRConfig::TSC)?;
+    let aperf_event = MSREvent::with_config(MSRConfig::APERF)?;    
+    let mperf_event = MSREvent::with_config(MSRConfig::MPERF)?;
+
+    let mut tsc = Builder::new(tsc_event)
+        .one_cpu(0)
+        .any_pid()
+        .enabled(true)
+        .exclude_hv(false)
+        .exclude_kernel(false)
+        .build()?;
+    let mut aperf = Builder::new(aperf_event)
+        .one_cpu(0)
+        .any_pid()
+        .enabled(true)
+        .exclude_hv(false)
+        .exclude_kernel(false)
+        .build()?;
+    let mut mperf = Builder::new(mperf_event)
+        .one_cpu(0)
+        .any_pid()
+        .enabled(true)
+        .exclude_hv(false)
+        .exclude_kernel(false)
+        .build()?;
+
+    std::thread::sleep(Duration::from_secs(1));
+    let tsc_val: u64 = tsc.read()?;
+    let ghz = tsc_val as f64 / (1000000000.0);
+    let aperf_val = aperf.read()?;
+    let mperf_val = mperf.read()?;
+    let ratio = aperf_val as f64 / mperf_val as f64;
+    let run_freq = ghz * ratio;
```

Comment:
For the values read here to be fully correct these counters should really be part of a group. Then they could be enabled and disabled together and the ratios calculated here would be fully correct.
```rust
let mut tsc = Builder::new(tsc_event)
    .one_cpu(0)
    .any_pid()
    .exclude_hv(false)
    .exclude_kernel(false)
    .build()?;
let mut aperf = Builder::new(aperf_event)
    .one_cpu(0)
    .any_pid()
    .exclude_hv(false)
    .exclude_kernel(false)
    .build_with_group(&mut tsc)?;
let mut mperf = Builder::new(mperf_event)
    .one_cpu(0)
    .any_pid()
    .exclude_hv(false)
    .exclude_kernel(false)
    .build_with_group(&mut tsc)?;
    
tsc.enable_group()?;
std::thread::sleep(Duration::from_secs(1));
tsc.disable_group()?;
```
In practice I don't think this matters that much but this is an example so it's better to get everything fully correct.